examples/vm_scheduling/rule_based_algorithm/first_fit.py

```diff
@@ -0,0 +1,39 @@
+from maro.simulator import Env
+from maro.simulator.scenarios.vm_scheduling.common import Action
+from maro.simulator.scenarios.vm_scheduling import AllocateAction, DecisionPayload, PostponeAction
+
+from algorithm import VMSchedulingAgent
+
+
+class FirstFit(VMSchedulingAgent):
+    def __init__(self):
+        super().__init__()
+        self._pm_list: list[int] = list()
+
+    def choose_action(self, decision_event: DecisionPayload, env: Env) -> Action:
+        valid_pm_num: int = len(decision_event.valid_pms)
+
+        # Check whether there exists a valid PM.
+        if valid_pm_num <= 0:
+            # No valid PM now, postpone.
+            action: PostponeAction = PostponeAction(
+                vm_id=decision_event.vm_id,
+                postpone_step=1
+            )
+        else:
+            # Use a valid PM based on its order.
+            chosen_idx: int = -1
+            for pm_idx in self._pm_list:
+                if pm_idx in decision_event.valid_pms:
```

Comment:
what is the difference between this implementation and choosing the valid pm with the smallest idx?